eng/testing/debug-dump-template.md

```diff
@@ -1,127 +1,146 @@
+Click the link to the dump on the `Helix Test Logs` tab in Azure DevOps. This is the same place you got these instructions from.
 
-## What is runfo?
+# Get the Helix payload
 
-Runfo is a dotnet global tool that helps get information about helix test runs and azure devops builds. For more information see [this](https://github.com/jaredpar/runfo/tree/master/runfo#runfo)
+[Runfo](https://github.com/jaredpar/runfo/tree/master/runfo#runfo) helps get information about helix test runs and azure devops builds. We will use it to download the payload and symbols:
+sos
+```script
+dotnet tool install --global runfo
+dotnet tool update --global runfo
+```
+If prompted, open a new command prompt to pick up the updated PATH.
+```script
+# On Windows
+runfo get-helix-payload -j %JOBID% -w %WORKITEM% -o %WOUTDIR%
+# On Linux and macOS
+runfo get-helix-payload -j %JOBID% -w %WORKITEM% -o %LOUTDIR%
+```
 
-### How do I install it?
+> NOTE: if the helix job is an internal job, you need to pass down a [helix authentication token](https://helix.dot.net/Account/Tokens) using the `--helix-token` argument.
 
-You just need to run:
+Now extract the files:
 
 ```script
-dotnet tool install --global runfo
-```
+# On Windows
+for /f %i in ('dir /s/b %WOUTDIR%\*zip') do tar -xf %i -C %WOUTDIR%
 
-If you already have it installed, make sure you have at least version `0.6.1` installed, which contains support to download helix payloads. If you don't have the latest version just run:
+# On Linux and macOS
+# obtain `unzip` if necessary; eg `sudo apt-get install unzip` or `sudo dnf install unzip`
+find %LOUTDIR% -name '*zip' -exec unzip -d %LOUTDIR% {} \;
+```
 
+Now use the [dotnet-sos global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos) to install the SOS debugging extension.
 ```script
-dotnet tool update --global runfo
+dotnet tool install --global dotnet-sos
+dotnet tool update --global dotnet-sos
+```
+If prompted, open a new command prompt to pick up the updated PATH.
+```script
+dotnet sos install --architecture Arm
+dotnet sos install --architecture Arm64
+dotnet sos install --architecture x86
+dotnet sos install --architecture x64
 ```
 
-## Download helix payload containing symbols:
+# Now choose a section below based on your OS.
 
-You can just achieve this by running:
+## If it's a Windows dump on Windows...
 
-```script
-runfo get-helix-payload -j %JOBID% -w %WORKITEM% -o <out-dir>
-```
+## ... and you want to debug with WinDbg
 
-> NOTE: if the helix job is an internal job, you need to pass down a [helix authentication token](https://helix.dot.net/Account/Tokens) using the `--helix-token` argument.
+Install or update WinDbg if necessary ([external](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools), [internal](https://osgwiki.com/wiki/Installing_WinDbg)). If you don't have a recent WinDbg you may have to do `.update sos`.
```

Comment:
Is there any advantage in using MS internal link to downloading windbg?

Reply:
Sometimes it has fixes that aren't on the stable release bits.